pyramid/viewderivers.py

```diff
@@ -499,18 +519,23 @@ def csrf_view(view, info):
     wrapped_view = view
     if enabled:
         def csrf_view(context, request):
-            if (
-                request.method not in safe_methods and
-                (
-                    # skip exception views unless value is explicitly defined
-                    getattr(request, 'exception', None) is None or
-                    explicit_val is not None
-                )
-            ):
+            if request.method not in safe_methods:
```

Comment:
Isn't this also missing a check for exceptions? or do you want to enable CSRF checks on exceptions always?

Because down below it is disabled unless it is explicitly set...